### PR TITLE
WINDUPRULE-617 Spring Web to quarkus rule

### PR DESCRIPTION
**Pull request by aureamunoz**
https://issues.redhat.com/browse/WINDUPRULE-617

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Can one of the admins verify this patch?

**Comment by jonathanvila** [comment]
Windup has a list of file names with mavan GAV , in order to find dependencies. 
The file you have used as test seems too new and could not be in that list.
Could you use an older version, please.

**Comment by aureamunoz** [comment]
I think this can be reviewed @jonathanvila

File: rules-reviewed/quarkus/springboot/springboot-web-to-quarkus.windup.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        <rule id="springboot-web-to-quarkus-00000">
+            <when>
+                <dependency groupId="org.springframework" artifactId="spring-web"/>
+            </when>
+            <perform>
+                <hint title="Replace the SpringBoot Web artifact with Quarkus 'spring-web' extension" effort="1" category-id="mandatory">
+                    <message>
+                        Replace the Spring Web artifact with Quarkus `spring-web` extension
+
+                        Spring Web is a spring-web artifact brought transitively by any `org.springframework.boot:spring-boot-*` dependency
```
Comment (jonathanvila) [inline rules-reviewed/quarkus/springboot/springboot-web-to-quarkus.windup.xml:26]: Are you sure spring-web is brought by *any* spring-boot-* artifact ?

Comment (aureamunoz) [reply rules-reviewed/quarkus/springboot/springboot-web-to-quarkus.windup.xml]: Thanks. Message changed in the last commit

File: rules-reviewed/quarkus/springboot/springboot-web-to-quarkus.windup.xml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        <rule id="springboot-web-to-quarkus-00000">
+            <when>
+                <dependency groupId="org.springframework" artifactId="spring-web"/>
+            </when>
+            <perform>
+                <hint title="Replace the SpringBoot Web artifact with Quarkus 'spring-web' extension" effort="1" category-id="mandatory">
+                    <message>
+                        Replace the Spring Web artifact with Quarkus `spring-web` extension
+
+                        Spring Web is a spring-web artifact brought transitively by any `org.springframework:spring-web*` dependency
```
Comment (jonathanvila) [inline rules-reviewed/quarkus/springboot/springboot-web-to-quarkus.windup.xml:26]: If you want to add a new line after "dependency" you should add 2 spaces at the end
```suggestion
                        Spring Web is a spring-web artifact brought transitively by any `org.springframework:spring-web*` dependency  
```

File: rules-reviewed/quarkus/springboot/tests/springboot-web-to-quarkus.windup.test.xml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0"?>
+<ruletest id="springboot-web-to-quarkus-tests"
+          xmlns="http://windup.jboss.org/schema/jboss-ruleset"
+          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <testDataPath>data/springboot-web</testDataPath>
+    <!-- By default sourceMode is True, so it will only consider .java files, not .jar binaries -->
+    <sourceMode>true</sourceMode>
```
Comment (jonathanvila) [inline rules-reviewed/quarkus/springboot/tests/springboot-web-to-quarkus.windup.test.xml:7]: Is this needed ?

Comment (aureamunoz) [reply rules-reviewed/quarkus/springboot/tests/springboot-web-to-quarkus.windup.test.xml]: no, I removed it

File: rules-reviewed/quarkus/springboot/springboot-web-to-quarkus.windup.xml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        <sourceTechnology id="springboot" />
+        <targetTechnology id="quarkus" />
+    </metadata>
+    <rules>
+        <rule id="springboot-web-to-quarkus-00000">
+            <when>
+                <dependency groupId="org.springframework" artifactId="spring-web"/>
+            </when>
+            <perform>
+                <hint title="Replace the SpringBoot Web artifact with Quarkus 'spring-web' extension" effort="1" category-id="mandatory">
```
Comment (jonathanvila) [inline rules-reviewed/quarkus/springboot/springboot-web-to-quarkus.windup.xml:22]: Shouldn't it be title and message aligned ? Title says `Replace the SpringBoot Web artifact` but message says `Replace the Spring Web artifact`

**Pull request merged**